shell/agents/Microsoft.Azure.Agent/Telemetry.cs

```diff
@@ -0,0 +1,218 @@
+using System.Text.Json;
+
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.WorkerService;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Azure.Agent;
+
+public class AzTrace
+{
+    private static readonly string s_installationId;
+    static AzTrace()
+    {
+        string azureConfigDir = Environment.GetEnvironmentVariable("AZURE_CONFIG_DIR");
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        string userProfilePath = Path.Combine(string.IsNullOrEmpty(azureConfigDir) ? userProfile : azureConfigDir, "azureProfile.json");
+
+        JsonElement array;
+        s_installationId = null;
+
+        if (File.Exists(userProfilePath))
+        {
+            using var jsonStream = new FileStream(userProfilePath, FileMode.Open, FileAccess.Read);
+            array = JsonSerializer.Deserialize<JsonElement>(jsonStream);
+            s_installationId = array.GetProperty("installationId").GetString();
+        }
+        else
+        {
+            try
+            {
+                Path.Combine(string.IsNullOrEmpty(azureConfigDir) ? userProfile : azureConfigDir, "azureProfile.json");
```

Comment:
This method is incorrect. This line is the same as the line 16, so I don't know why you are doing this twice.

Reply:
Pushed a commit to reslove this. Added `AzureRmContextSettings.json`